### PR TITLE
Migrate security/initializable

### PR DESCRIPTION
**Pull request by andrew-fleming**
Fixes (partial) #550 

#### PR Checklist

- [x] Tests
- [ ] Tried the feature on a public network
- [ ] Documentation

This PR proposes to change the view func `initialized` to `is_initialized` because it's more explicit and easier to read:
`initialize`/`is_initialized` vs `initialize`/`initialized`.

Supersedes #588.

### CODE REVIEW

File: src/openzeppelin/security/initializable.cairo [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#[contract]
+mod Initializable {
+    struct Storage {
+        _initialized: bool,
```
Comment (martriay) [inline src/openzeppelin/security/initializable.cairo:4]:
```suggestion
        initialized: bool,
```
any reason not to?

**Review by martriay — APPROVED** [review]
good work!

**Comment by milancermak** [comment]
Hey, I see you guys using `#[internal]`, I haven't seen it elsewhere so far. If you don't mind educating me, what does it do?

**Comment by andrew-fleming** [comment]
Hey, @milancermak! The `#[internal]` doesn't do anything, it just makes the contracts a bit more readable with how we're currently structuring them. See https://github.com/OpenZeppelin/cairo-contracts/pull/584#discussion_r1147522771

**Pull request merged**